src/main/java/org/broadinstitute/hellbender/tools/copynumber/denoising/SVDDenoisingUtils.java

```diff
@@ -230,15 +230,15 @@ private static PreprocessedStandardizedResult preprocessPanel(final RealMatrix r
             logger.info(String.format("A value of 100 was provided for argument %s, so the corresponding filtering step will be skipped...",
                     CreateReadCountPanelOfNormals.MAXIMUM_ZEROS_IN_SAMPLE_PERCENTAGE_LONG_NAME));
         } else {
-            logger.info(String.format("Filtering samples with a fraction of zero-coverage intervals above %.2f percent...", maximumZerosInSamplePercentage));
-            final int maxZerosInSample = calculateMaximumZerosCount(countNumberPassingFilter(filterIntervals), maximumZerosInSamplePercentage);
+            logger.info(String.format("Filtering samples with a fraction of zero-coverage intervals greater than or equal to %.2f percent...", maximumZerosInSamplePercentage));
+            final int numPassingIntervals = countNumberPassingFilter(filterIntervals);
             IntStream.range(0, numOriginalSamples)
                     .filter(sampleIndex -> !filterSamples[sampleIndex])
                     .forEach(sampleIndex -> {
-                        final int numZerosInSample = (int) IntStream.range(0, numOriginalIntervals)
+                        final double numZerosInSample = (double) IntStream.range(0, numOriginalIntervals)
                                 .filter(intervalIndex -> !filterIntervals[intervalIndex] && readCounts.getEntry(sampleIndex, intervalIndex) == 0.)
                                 .count();
-                        if (numZerosInSample > maxZerosInSample) {
+                        if (numZerosInSample / numPassingIntervals >= maximumZerosInSamplePercentage / 100.) {
```

Comment:
I'm a little confused, why is numZerosInSample a double rather than an int?
If you need it to be a double so that the fraction is a double, why not cast at the point of computing the fraction?

Reply:
In both cases, the cast happens in the next line and the variable is not used elsewhere, so I'm OK keeping it like this. Certainly it's valid to represent an integer with a double, at least here...?

Reply:
@samuelklee It's fine the way it is.